modules/searchbar/animated.tsx

```diff
@@ -1,137 +1,104 @@
 import * as React from 'react'
 
-import {StyleSheet, Animated} from 'react-native'
+import {StyleSheet, Animated, useWindowDimensions} from 'react-native'
 import {white} from '@frogpond/colors'
 import {SearchBar} from './searchbar'
-import {Viewport} from '@frogpond/viewport'
-
-type Props = {
-	value: string
-	active: boolean
-	placeholder?: string
-	onSubmit?: () => unknown
-	onFocus?: () => unknown
-	onCancel?: () => unknown
-	onChange?: (value: string) => unknown
-	title?: string
+import {Props, AnimatedValueType} from './types'
+
+let headerOpacitySpec: AnimatedValueType = {start: 1, end: 0, duration: 200}
+let searchBarTopSpec: AnimatedValueType = {start: 71, end: 10, duration: 200}
+let containerHeightSpec: AnimatedValueType = {
+	start: 125,
+	end: 64,
+	duration: 200,
 }
 
-export class AnimatedSearchBar extends React.Component<Props> {
-	componentDidUpdate(prevProps: Props) {
-		if (this.props.active !== prevProps.active) {
-			if (this.props.active) {
-				this.activateSearch()
-			} else {
-				this.deactivateSearch()
-			}
-		}
-	}
+const headerOpacity = new Animated.Value(headerOpacitySpec.start)
+const searchBarTop = new Animated.Value(searchBarTopSpec.start)
+const containerHeight = new Animated.Value(containerHeightSpec.start)
```

Comment:
Does this work? I worry that the instances of Animated.Value need to be inside the render, or they'll be shared between instances of the component?

Reply:
I'll check if it works. If it does not work, we will need to revert back to the hooks version of these, then.

Reply:
Let's do that for these three?